zulipterminal/ui_tools/views.py

```diff
@@ -775,6 +777,55 @@ def __init__(self, controller: Any, widgets: List[Any],
         self.log = urwid.SimpleFocusListWalker(widgets)
         super().__init__(self.log)
 
+    def max_widths(self, contents: PopUpViewContent, title_len: int,
+                   dividechars: int) -> List[int]:
+        """
+        Returns maximum width required per column and sets width accordingly.
+        """
+        # Add +6 to compensate for the surrounding borders.
+        heading_width = title_len + 6
+        row_widths = []
+        for category, content in contents:
+            heading_width = max(len(category), heading_width)
+            for row in content:
+                if isinstance(row, str):
+                    heading_width = max(len(row), heading_width)
+                elif isinstance(row, tuple):
+                    row_widths.append(list(map(len, row)))
+        col_max_widths = [max(row_width) for row_width in zip(*row_widths)]
+        self.width = max(heading_width, sum(col_max_widths) + dividechars)
+        return col_max_widths
+
+    def build_widgets(self, contents: PopUpViewContent, title_len: int,
+                      dividechars: int=2) -> List[Any]:
+        """
+        Return a list of widgets to be rendered and sets height accordingly.
+        """
+        col_max_widths = self.max_widths(contents, title_len, dividechars)
```

Comment:
You're passing these values in (contents, title_len, dividechars), to use them in an interior method and only contents otherwise. How might this look if you didn't nest the calls, but called them sequentially?